library/core/src/array/mod.rs

```diff
@@ -959,3 +959,32 @@ where
     let mut map = iter.map(NeverShortCircuit);
     try_collect_into_array(&mut map).map(|NeverShortCircuit(arr)| arr)
 }
+
+/// Tries to create an array `[T; N]` from the items of `iter`. If and only if `iter` holds
+/// `>= N` items it returns `OK([T; N])`. Otherwise, `Err(IntoIter<T; N>)` is returned.
+///
+/// # Arguments
+///
+/// * `iter`: Type that implements `IntoIterator` where `Items = T`.
+///
+/// # Example
+///
+/// ```rust
+/// #![feature(try_from_iter)]
+/// let vec = vec![0, 1, 2, 3];
+/// let array = core::array::try_from_iter::<_, 4>(vec);
+/// assert!(array.is_ok());
+/// assert_eq!(array.unwrap(), [0, 1, 2, 3]);
+///
+/// let vec = vec![0, 1, 2];
+/// let array = core::array::try_from_iter::<_, 4>(vec);
+/// assert!(array.is_err());
+/// ```
+#[unstable(feature = "try_from_iter", issue = "none")]
+#[inline]
+pub fn try_from_iter<I, const N: usize>(iter: I) -> Result<[I::Item; N], IntoIter<I::Item, N>>
```

Comment:
This consumes `iter`, while `next_chunk` takes a `&mut iter`. Those are somewhat different semantics because it raises the question what happens when the remainder of the iterator could yield additional elements.

Reply:
yess that is true! i was wondering about this as well when choosing the API layout, but ended up going with taking ownership of `iter`, as it seems to be the standard behavior of `from` functions and is probably what users would expect. Your point is still very valid, as most other collections that can be created through some form of `from`, `From`, `try_from`, or `TryFrom` don't have this size constraint that `[T; N]` has. 

I'll add this to the RFC under **Alterntives**; yet, if this `&mut self` ends up making more sense then changing it to that should not be to difficult!

Reply:
One could also maybe imagine returning something like `Result<([T; N], IntoIterator::Iterator), IntoIter<T, N>>` but that might be a bit to contrived?